### PR TITLE
 #1537 Nuget package with specified size

### LINKED ISSUE #1537: Implement support for generating test artifacts with a specified size for Nuget
# Task Description

We need to be able to create test artifacts with a specified size.

# Classes of Interest

* `org.carlspring.strongbox.artifact.generator.NugetArtifactGenerator`
* `org.carlspring.strongbox.testing.artifact.NugetTestArtifact`

# Tasks

The following tasks will need to be carried out:
* [ ] Apply the necessary changes to the `org.carlspring.strongbox.artifact.generator.NugetArtifactGenerator`.
* [ ] Apply the necessary changes to the `org.carlspring.strongbox.testing.artifact.NugetTestArtifact`.
* [ ] Add tests for this.

# Task Relationships

This task:
* Is a sub-task of: #1534 
* Relates to: #1535 , #1536 , #1538 , #1540 

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @fuss86
  * @sbespalov
  * @steve-todorov

### PR DESCRIPTION
**Pull request by wpater**
* Add size property in `NugetTestArtifact` annotation
* Update `NugetArtifactGenerator` to use size property

# Pull Request Description

This pull request closes #1537 

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [x] Yes, please see ~~#1536~~
  * [ ] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

### CODE REVIEW

**Comment by fuss86** [comment]
It looks like https://github.com/strongbox/strongbox/pull/1559 should be merged first. Then we will review this PR.

**Comment by carlspring** [comment]
#1536 has been merged.

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/NugetArtifactGenerator.java [hunk 1]
```diff
@@ -54,6 +56,8 @@
 
     private static final String PACKAGING_NUPKG = "nupkg";
 
+    private static final int DEFAULT_SIZE = 1000000;
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/NugetArtifactGenerator.java]: Similarly to the previous pull request, could you please change this:
```
    private static final int DEFAULT_SIZE = 1000000;
```
 to:
```
    private static final int DEFAULT_BYTE_SIZE = 1000000;
```
?

Thanks! :)

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/NugetArtifactGenerator.java [hunk 2]
```diff
@@ -72,59 +76,60 @@ public Path generateArtifact(String id,
                                  long size)
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/NugetArtifactGenerator.java]: Could you please rename these variable to `byteSize`? Thanks! :)

**Comment by wpater** [comment]
> Could you please change `size` to `byteSize`? Thanks! :)

Changed to `bytesSize`.

**Comment by carlspring** [comment]
Thanks! I'll trigger the Jenkins build pipeline.

**Comment by carlspring** [comment]
@wpater ,

Would you mind rebasing? Thanks! :)

**Comment by wpater** [comment]
> @wpater ,
> 
> Would you mind rebasing? Thanks! :)

Done.

**Comment by carlspring** [comment]
I will merge it once the build is green. Thanks! :)

**Comment by carlspring** [comment]
Thanks @wpater ! :)

**Pull request merged**